torchtune/data/_prompt_templates.py

```diff
@@ -226,3 +242,29 @@ def __call__(
 
 Please see :class:`~torchtune.data.PromptTemplate` for full API arguments.
 """
+
+
+def _get_prompt_template(
```

Comment:
This is super cool - we'll need some examples to make it clear to the user all the ways this can work.